File: programmers/연습문제/가장 긴 팰린드롬.py
def solution(s):
    answer = []
    for i in range(len(s) - 1):
        for j in range(i + 2, len(s) + 2):
            origin = s[i: j]
            if origin == origin[::-1]:
                answer.append(len(origin))
                print(origin)
    if len(answer) == 0 or len(s) == 1:
        return 1
    return max(answer)

File: programmers/코딩테스트 연습/연습문제/가장 긴 팰린드롬.cpp
#include <bits/stdc++.h>
using namespace std;
bool isPalindrome(string s){
    for(int i = 0; i < s.size() / 2; i++){
        if(s[i] != s[s.size()-i-1]) return false;
    }
    return true;
}

int solution(string s){
    int answer=0;
    int cnt = 0;
    for(int l = 1; l <= s.size(); l++){
        for(int i = 0; i < s.size(); i++){
            string tmp ="";
            for(int j = i; i+l <= s.size() && j < i+l; j++){
                tmp += s[j];
            }
            int size = tmp.size();
            if(isPalindrome(tmp)) answer = max(answer,size);
        }
    }

    return answer;
}